enclave/trust_requirement.go

```diff
@@ -0,0 +1,279 @@
+package enclave
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	enclaveTrustRequirement "github.com/enclave-networks/go-enclaveapi/data/trustrequirement"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type trustRequirementResourceType struct{}
+
+func (t trustRequirementResourceType) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Attributes: map[string]tfsdk.Attribute{
+			"id": {
+				Type:     types.Int64Type,
+				Computed: true,
+			},
+			"description": {
+				Type:     types.StringType,
+				Required: true,
+			},
+			"notes": {
+				Type:     types.StringType,
+				Optional: true,
+			},
+			"user_authentication": {
+				Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+					"authority": {
+						Type:     types.StringType,
+						Required: true,
+					},
+					"tenant_id": {
+						Type:     types.StringType,
+						Optional: true,
+					},
+					"group_id": {
```

Comment:
In order to spare us future updates every time the UI changes to add another 'claim' value, I think it may be better here to directly expose a list of claims `{ claim: "groups", value: "" }` to check against rather than explicitly having a `group_id` field.

I know that's sort of a blend of how the UI works vs how the API works, but it feels like it may be best for terraform; it makes the terraform stuff much more powerful that way as well, otherwise I'm going to have to ask you to add properties here all the time.

If we do this, then it also means that the `claims` property can sit adjacent to the authority, rather than underneath it.

(Note, we do need to add a design for the portal to surface claims added via the API anyway that aren't in the UI).